src/test_reconciler.js

```diff
@@ -105,6 +105,11 @@ export default class TestReconciler {
       return 'New snapshot is ready to write';
     }
 
+    // Improve @testing-library/react error messages handling
+    if (string.startsWith('Error: Unable to find an element with the text')) {
+      return string;
+    }
+
```

Comment:
Interesting idea @jmarceli. It seems like we're missing an extension point either deciding what to keep or deciding what to show. I've been out of the fold for a while. Perhaps @connectdotz or @stephtr have a strong opinion about our architectural direction to address this.

Reply:
I'm aware that such message based decisions are not future proof but I didn't find any other way to address library specific tests and current short message is not very user friendly. If I'm missing something just let me know.

Reply:
It seems fine for me and it will work in my case but I'm not sure how would you like to detect standard Jest message format?
In your link the line `message.split('   at', 1)[0].trim()` is highlighted but I have no idea how it can be used for standard Jest messages detection (as far as I can tell @testing-library messages also has `'   at'` part).
Based on examples provided in the other tests I've decided to go with a different way of detecting standard Jest messages.
@connectdotz please let me know what do you think about that and if it's required how would you like to modify that.

Reply:
> but I'm not sure how would you like to detect standard Jest message format?

that's a good question, you are right that it is hard to say definitively... Maybe let's tackle it from another angle. 

Looking at the original algorithm, I think the intent of this short message is to show the "Expected vs Received" information and by-passing stack trace. So to keep this goal in mind, we can ask ourselves if we can produce this "target" message format, if not, then we are better off to just use the full message...

what do you think?

Reply:
I came up with this not-so-clever regex `/expected[\s\S]*received/i` please let me know if you have something more sophisticated in your mind. For the current test suite, it is working fine.

Reply:
Even though I like this approach, I see the issue of false positives as soon as a custom string contains the two words. We probably won't get it to work perfectly (which isn't necessary), but I would at least precise it to `/Expected: .*\nReceived: /s` (and hope, that Jest is never going to localize their messages 😉).

Reply:
I'm not sure about `/Expected: .*\nReceived: /s` as it won't work for current test suite. I mean this message `Expected value to be falsy, instead received\n  2`. The only improvement I can make would be `/Expected[\s\S]*[r|R]eceived/s` but I don't think it is an improvement worth adding.

Reply:
You are completely correct, I didn't have that in mind.